src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -397,7 +397,7 @@ static bool updateAttribute(BSONObj& attrs, BSONObjBuilder* toSet, ContextAttrib
   actualUpdate = false;
 
   /* Attributes with metadata ID are stored as <attrName>__<ID> in the attributes embedded document */
-  std::string effectiveName = caP->name;
+  std::string effectiveName = dbDotEncode(caP->name);
```

Comment:
At the end, given that these fucntions are highly coupled with "dot semantcis" I have included "Dot" in the name. If in the future they become more generic, name could be changed.